### ISSUE TITLE
Recovery after loss of internet connection fails

### ISSUE THREAD

**Issue by Victov** [octokit/octokit.net#1590]
Github API does not recover from losing internet connection once, even though the connection has been restored in the meantime.

**Reproduction steps**
1. Put your device in airplane mode (Tested on windows 10)
2. Use the API to get data from github
3. Catch the exception that is being thrown due to lack of connection
4. Disable airplane mode on your device
5. Request data from github again
6. Watch how exceptions are still being thrown, even though internet connection has been restored.

**Comment by ryangribble** [comment]
This sounds really weird.

Can you post more details (eg the type of application and whether this is during a single application life cycle or across separate program executions) and also the exception and stack trace details of the initial exception and also the one "after connection restored"?

**Comment by Victov** [comment]
This happens in one life cycle. Restarting the program after the internet connection is makes everything function perfectly again. It's a very basic console application that takes some data from github and interprets it for some data analytics purpose. It all works perfectly fine when the internet connection is never interupted.

The problem seems to be an AuthorizationException. I tried setting the credentials again when connection was restored, but that did not fix the problem. 
I also tried re-creating the entire GithubClient object when internet connection was restored, but that also did not fix the problem. See the attached screenshot.

![image](https://cloud.githubusercontent.com/assets/6769990/25390649/6fdb0ca8-29d4-11e7-9889-2c0dd7d5a419.png)
